Shared/Extensions/URLExtensions.swift

```diff
@@ -421,7 +429,15 @@ extension URL {
     }
 
     public var originalURLFromErrorURL: URL? {
-        let components = URLComponents(url: self, resolvingAgainstBaseURL: false)
+        var url = self
```

Comment:
Is it possible to add unit tests for `decodeReaderModeURL` and `originalURLFromErrorURL` ?

Reply:
Test are already tests: `testoriginalURLFromErrorURL`
If any of the tests currently fail, it means this PR is invalid because we want all the tests to still work even with the changes. That way, we know for sure we didn't break anything. It should work on iOS 13+, 14+ and 15+.